File: java/src/tugas_pbo/tugas3/Persegi.dart
import 'BangunDatar.dart';

class Persegi extends BangunDatar {
  // Menggunakan constructor 1 parameter dari superclass
  Persegi(double sisi) : super.satuParameter(sisi);

  @override
  double hitungLuas() {
    return varA * varA;
  }

  @override
  void tampilkan() {
    print("Persegi");
    print("==========================");
    print("Sisi: $varA");
    print("Luas: ${hitungLuas()}");
    print("==========================");
  }
}

File: java/src/latihan_pbo/abstrak/Persegi.java
package latihan_pbo.abstrak;

public class Persegi extends BangunDatar{
    private double sisi;

    public Persegi (double sisi){
        super("Persegi");
        this.sisi = sisi;
    }

    @Override
    double hitungLuas(){
        return sisi * sisi;
    }
}
